infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java

```diff
@@ -484,6 +514,24 @@ public EndpointMetaDataBuilder queryListParam(final ParameterMetadata<?> paramet
       return this;
     }
 
+    public EndpointMetaDataBuilder requiredHeader(final ParameterMetadata<?> headerMetadata) {
```

Comment:
NIT: Following `queryParam(..)` and `queryParamRequired(..)` naming, we would call this `headerRequired(..)`.